spec/lib/smb2/packet_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'smb2'
+
+RSpec.describe Smb2::Packet do
+  subject(:packet) do
+    klass = Class.new(described_class)
+    klass.instance_eval do
+      nest :header, Smb2::Packet::RequestHeader
+      # struct_size is part of the API and must be present
+      # 2 bytes for struct_size itself, 2 for data_offset, 4 for data_length
+      unsigned :struct_size, 16, default: 2 + 2 + 4
+      data_buffer :data, 32
+      rest :buffer
+    end
+    klass.new
+  end
+
+  it { is_expected.to respond_to(:recalculate) }
```

Comment:
Please keep your respond_to expectations ordered by message name so it's easier to spot if one is missing.